networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -147,30 +150,61 @@ where
 			}
 		}
 
-		let length = transactions.len();
-		if length > 0 {
-			let mut light_node_client = self.light_node_client.clone();
-			let span = info_span!(
+		if transactions.len() > 0 {
+			info!(
 				target: "movement_timing",
-				"batch_write",
 				batch_id = %batch_id,
-				length = length
+				transaction_count = transactions.len(),
+				"built_batch_write"
 			);
-			light_node_client
-				.batch_write(BatchWriteRequest { blobs: transactions })
-				.instrument(span)
-				.await?;
-			// We now consider the transactions no longer in mempool flight.
-			self.executor.decrement_transactions_in_flight(length as u64);
+			let batch_write = BatchWriteRequest { blobs: transactions };
+			sender.send(batch_write).await?;
 		}
 
 		Ok(())
 	}
 
+	pub async fn send_transaction_batch_writes(
+		&self,
+		sender: Sender<BatchWriteRequest>,
+	) -> Result<(), anyhow::Error> {
+		loop {
+			self.next_transaction_batch_write(sender.clone()).await?;
```

Comment:
Useless clone:
```suggestion
			self.next_transaction_batch_write(sender).await?;
```

Reply:
The value is moved in the previous iteration of the loop. Do you want me to return the sender?

Reply:
Sorry, I missed the loop when I looked at it yesterday.
But since `next_transaction_batch_write` is a helper method and borrows `self` anyway, it could borrow `sender` as well? These are all micro-optimizations unlikely to have major impact, anyway.